### PR TITLE
Replaced @std/esm with rollup and babel

### PR DESCRIPTION
**Pull request by taras**
`@std/esm` was causing a lot of problems when consuming this package. It wasn't clear how `@std/esm` should be included in different scenarios. This package introduces rollup for building distributable package and babel for running tests.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,22 +2,34 @@
   "name": "funcadelic",
   "version": "0.3.0",
   "description": "Functional Fundamentals are Fun!",
-  "main": "nodule.js",
-  "module": "src/funcadelic.js",
+  "main": "dist/funcadelic.cjs.js",
+  "module": "dist/funcadelic.esm.js",
```
Comment (cowboyd) [inline package.json]: Any reason not to point `module` at `src/funcadelic.js`? It feels like it ought to be possible to include funcadelic in a build system that has never been touched by `rollup`.  In other words, it feels natural that there should be _some_ entry point specified for the pure source.

File: package.json [hunk 2]
```diff
@@ -2,22 +2,34 @@
...
+  "browser": "dist/funcadelic.umd.js",
   "repository": "https://github.com/cowboyd/funcadelic.js.git",
   "author": "Charles Lowell <cowboyd@frontside.io>",
   "license": "MIT",
   "private": false,
   "scripts": {
-    "test": "mocha tests/index.js"
+    "build": "rollup -c",
```
Comment (cowboyd) [inline package.json]: I'm a fan of using short options from the CLI, but full options in scripts. That way it's optimized for readibility. Let's expand this to `rollup --config`

File: .babelrc [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "presets": [
+    "es2015"
```
Comment (cowboyd) [inline .babelrc]: I don't understand babel presets. Isn't it recommended to not use a year, and instead use `env`?

Comment (taras) [reply .babelrc]: Yeah, I keep track of these things. But I'll env instead

File: package.json (hunk 2 above)
Comment (taras) [reply package.json]: ok

File: package.json (hunk 1 above)
Comment (taras) [reply package.json]: rollups creates one file for all modules which is faster to parse and easier to navigate as a dependency. It seems like a reasonable default. I'll leave it as is if you don't object.

**Comment by taras** [comment]
@cowboyd applied the changes.

**Pull request merged**